File: java/src/main/java/com/epam/eng/pattern/interpreter/PhoneNumberMasker.java
package com.epam.eng.pattern.interpreter;

public class PhoneNumberMasker {
    public static void main(String[] args) {
        String phoneNumber = "123-456-7890";
        String maskedPhoneNumber = maskPhoneNumber(phoneNumber);
        System.out.println(maskedPhoneNumber);
    }

    public static String maskPhoneNumber(String phoneNumber) {
        String[] parts = phoneNumber.split("-");
        String firstPart = parts[0];
        String secondPart = parts[1];
        String thirdPart = parts[2];
        String maskedSecondPart = "XXX";
        return firstPart + "-" + maskedSecondPart + "-" + thirdPart;
    }
}

File: ts/interpreter/PhoneNumberMasker.ts
class PhoneNumberMasker {

  public maskPhoneNumber(phoneNumber: string): string {
    const parts = phoneNumber.split("-");
    const firstPart = parts[0];
    const secondPart = parts[1];
    const thirdPart = parts[2];
    const maskedSecondPart = "XXX";
    return `${firstPart}-${maskedSecondPart}-${thirdPart}`;
  }
}

const phoneNumber = "123-456-7890";
const maskedPhoneNumber = new PhoneNumberMasker().maskPhoneNumber(phoneNumber);
console.log(maskedPhoneNumber);

// todo: consider to create an Interpreter can interpret expression ###-XXX-###, transfer phone number from 123-456-7890 to 123-XXX-7890

